noir-projects/aztec-nr/aztec/src/messages/encryption/aes128.nr

```diff
@@ -125,18 +140,23 @@ fn extract_close_to_uniformly_random_256_bits_from_ecdh_shared_secret_using_sha2
     sha256::digest(shared_secret_bytes_with_separator)
 }
 
-fn derive_aes_symmetric_key_and_iv_from_ecdh_shared_secret(
-    shared_secret: Point,
-    randomness_extraction_fn: fn(Point) -> [u8; 32],
-) -> ([u8; 16], [u8; 16]) {
-    let random_256_bits = randomness_extraction_fn(shared_secret);
-    let mut sym_key = [0; 16];
-    let mut iv = [0; 16];
-    for i in 0..16 {
-        sym_key[i] = random_256_bits[i];
-        iv[i] = random_256_bits[i + 16];
+fn derive_aes_symmetric_key_and_iv_from_uniformly_random_256_bits<let N: u32>(
+    many_random_256_bits: [[u8; 32]; N],
+) -> [([u8; 16], [u8; 16]); N] {
+    // Many (sym_key, iv) pairs:
+    let mut many_pairs: [([u8; 16], [u8; 16]); N] = std::mem::zeroed();
+    for k in 0..N {
```

Comment:
No limit on `N` here?

Reply:
Upon returning to this after 12 years away from it, I'm not sure why it would be limited. Make all the pairs you want! Go nuts! 2^32 pairs. Beautiful. Casually create 4 billion ciphertexts.